devtools/src/main/scala/kalix/devtools/impl/DevModeSettings.scala

```diff
@@ -44,6 +46,60 @@ object DevModeSettings {
 
   }
 
+  private def grpcClientConfig(serviceName: String, host: String, port: Int): Config =
+    ConfigFactory.parseString(s"""
+         |akka.grpc.client.$serviceName {
+         |  service-discovery {
+         |    service-name = "$serviceName"
+         |  }
+         |  host = "$host"
+         |  port = $port
+         |  use-tls = false
+         |}
+         |""".stripMargin)
+
+  private def restClientConfig(serviceName: String, host: String, port: Int) =
+    ConfigFactory.parseString(s"""${portMappingsKeyPrefix}.$serviceName="$host:$port"""")
+
+  def addDevModeConfig(mainConfig: Config): Config = {
+    // enrich config with extra dev-mode service port mappings if applicable
+    DockerComposeUtils
+      .fromConfig(mainConfig)
+      .map { dcu =>
+        // read user-function port from docker-compose and overwrite the main config
+        val adaptedConfig =
+          ConfigFactory
+            .parseString(s"kalix.user-function-port = ${dcu.userFunctionPort}")
+            .withFallback(mainConfig)
+
+        dcu.servicesHostAndPortMap.foldLeft(adaptedConfig) { case (main, (serviceName, hostAndPort)) =>
+          // DockerComposeUtils (dcu) is only available when running locally,
+          // therefore should always use localhost (0.0.0.0)
+          val host = "0.0.0.0"
+          val port = HostAndPort.extractPort(hostAndPort)
+          // when configuring through DockerComposeUtils, we need to parts.
+          // a config for a gRPC client and a direct config for REST WebClient
+          main
+            .withFallback(grpcClientConfig(serviceName, host, port)
+              .withFallback(restClientConfig(serviceName, host, port)))
+        }
+      }
+      .getOrElse {
+        // when not relying on DockerComposeUtils, we should still look for dev-mode settings
+        // and translate them to gRPC clients
+        if (mainConfig.hasPath(portMappingsKeyPrefix)) {
+          val portMappings =
+            mainConfig.getConfig(portMappingsKeyPrefix).entrySet().asScala
+          portMappings.foldLeft(mainConfig) { case (main, entry) =>
+            val (host, port) = HostAndPort.extract(entry.getValue.unwrapped().toString)
+            main.withFallback(grpcClientConfig(entry.getKey, host, port))
+          }
+
+        } else
+          mainConfig
+      }
+  }
```

Comment:
This is the missing part that was making the build to fail. 

It goes as such:

If running locally (exec Main) or through plugin, we read the docker-compose.yml file to learn where the other services are running. That means that we are running in a non-dockerized environment and we need to reach whatever other services' proxy through localhost (0.0.0.0).

So, first we read the file and build the mappings. Then we need to create gRPC client configs and web client configs for each of them. But in both cases, we need to ignore the host declared in the file and replace it with 0.0.0.0. 

On the other hand, if we run the user function from inside a container, we need to have a full host:port setup. In such a case, the port mapping will need to be something like: 
`kalix.dev-mode.service-port-mappings.foo=host.docker.internal:9001`. Where `foo` is the name of the service an `host.docker.internal` the address to access (intra container calls).

Note that the settings is initially target to the proxy. Between two proxy, we still need to use `host.docker.internal`.